snow/engine/snowman/bootstrap/interval/tree_test.go

```diff
@@ -0,0 +1,389 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package interval
+
+import (
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/database/memdb"
+)
+
+func newTree(require *require.Assertions, db database.Database, intervals []*Interval) *Tree {
```

Comment:
nit: We could add a variadic `intervals Interval...` parameter at the end of `interval.NewTree` to avoid this extra testing function

Reply:
I'd prefer not to introduce functionality to the production code that is only used during testing